futures-util/src/try_future/try_join_all.rs

```diff
@@ -0,0 +1,182 @@
+//! Definition of the `TryJoinAll` combinator, waiting for all of a list of
+//! futures to finish with either success or error.
+
+use std::fmt;
+use std::future::Future;
+use std::iter::FromIterator;
+use std::mem;
+use std::pin::Pin;
+use std::prelude::v1::*;
+use std::task::Poll;
+use super::TryFuture;
+
+#[derive(Debug)]
+enum ElemState<F>
+where
+    F: TryFuture,
+{
+    Pending(F),
+    Done(Option<Result<F::Ok, F::Error>>),
```

Comment:
Since this is shortcircuiting errors, I don't think `ElemState::Done(Some(Err(...)))` is ever used? You could simplify this to `Done(Option<F::Ok>)`.

Reply:
Good point, that's certainly correct. I'll make the change real quick.